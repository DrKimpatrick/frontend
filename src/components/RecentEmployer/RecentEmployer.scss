.recent-employer-section {
  position: relative;
  .title {
    font-size: 22px;
    color: #747474;
    text-decoration: none solid rgb(116, 116, 116);
  }

  .back-arrow {
    position: absolute;
    color: #747474;
    left: -43px;
    top: 0px;
    font-size: 22px;
  }

  .input-height {
    height: 40px !important;
    border-radius: 4px;
    color: #747474;
    &::placeholder {
      color: #c9c9c9;
    }
  }

  .input-height-input {
    color: #747474;
    &::placeholder {
      color: #c9c9c9;
    }
  }
}

@media (max-width: 786px) {
  .recent-employer-section {
    width: 60%;

    .back-arrow {
      display: none;
    }
  }
}

@media (max-width: 500px) {
  .recent-employer-section {
    width: 90%;

    .title {
      text-align: center;
    }

    .back-arrow {
      display: none;
    }
  }
}
