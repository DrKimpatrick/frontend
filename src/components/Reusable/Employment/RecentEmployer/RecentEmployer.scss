.recent-employer-section {
  position: relative;
  .title {
    font-size: 22px;
    color: #747474;
    text-decoration: none solid rgb(116, 116, 116);
  }

  .back-arrow {
    position: absolute;
    color: #747474;
    left: -43px;
    top: 0px;
    font-size: 22px;
  }

  .input-height {
    height: 42px !important;
    border-radius: 4px;
    color: #747474;
    &::placeholder {
      color: #c9c9c9;
    }
  }

  .input-height-input {
    color: #747474;
    &::placeholder {
      color: #c9c9c9;
    }
  }
}

@media (max-width: 786px) {
  .recent-employer-section {
    padding: 10px !important;

    .containers {
      width: 60%;
    }

    .back-arrow {
      display: none;
    }

    .title {
      width: 100% !important;
      text-align: center;
    }
  }
}

@media (max-width: 500px) {
  .recent-employer-section {
    margin-bottom: 40px !important;
    .containers {
      width: 100%;
    }
  }
}
