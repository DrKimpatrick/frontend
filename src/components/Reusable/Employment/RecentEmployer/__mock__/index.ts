export const initialState = {
  users: {
    currentUser: {
      isLoggedIn: true,
      loading: false,
      error: null,
      message: null,
      created: false,
      data: {
        firstName: 'John',
        lastName: 'Smith',
        profilePicture:
          'https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_960_720.png'
      }
    },
    user: {
      _id: 'id',
      username: 'username',
      email: 'email',
      roles: ['talent'],
      updatedAt: '2020-10-10',
      dateRegistered: '2020-10-10',
      paymentStatus: 'verified',
      featureChoice: 'local',
      verified: true
    }
  },
  employments: {
    employment: {
      companyName: 'snow',
      supervisor: 'supervisor',
      title: 'title',
      startDate: '2020-10-5',
      responsibilities: ['res'],
      accomplishments: ['res'],
      favoriteProject: 'res',
      _id: 'id'
    },
    errors: null,
    loading: false,
    submitLoading: false
  },
  messages: {
    message: 'sent'
  },
  educations: {
    education: {
      schoolName: 'schoolName'
    }
  },
  skills: {
    userSkill: [
      {
        skill: {
          _id: '5fcf1c5b11c16c41f7c6c723',
          skill: 'Python',
          createdAt: '2020-12-08T06:25:31.128Z',
          updatedAt: '2020-12-08T06:25:31.128Z'
        },
        level: 'intermediate',
        verificationStatus: 'verified',
        _id: '5fcf52105cd56b9843252696',
        createdAt: '2020-12-08T10:14:40.231Z',
        updatedAt: '2020-12-08T13:05:46.873Z'
      }
    ]
  }
};
