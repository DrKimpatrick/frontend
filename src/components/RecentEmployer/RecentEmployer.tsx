import React, { FC } from 'react';
import { withRouter } from 'react-router-dom';
import './RecentEmployer.scss';
import ArrowBackTwoToneIcon from '@material-ui/icons/ArrowBackTwoTone';
import ArrowRightAltTwoToneIcon from '@material-ui/icons/ArrowRightAltTwoTone';
import Select from 'components/SelectOption/SelectOption';
import DatePicker from 'components/DatePicker/DatePicker';

type props = {};
const RecentEmployer: FC<props> = (props: any) => {
  const options = [
    { value: 'Software Engineer', label: 'Software Engineer' },
    { value: 'Product Manager', label: 'Product Manager' },
    { value: 'Product Designer', label: 'Product Designer' },
    { value: 'Software Engineer1', label: 'Software Engineer1' },
    { value: 'Product Manager1', label: 'Product Manager1' },
    { value: 'Product Designer1', label: 'Product Designer1' },
    { value: 'Software Engineer2', label: 'Software Engineer2' },
    { value: 'Product Manager2', label: 'Product Manager2' },
    { value: 'Product Designer2', label: 'Product Designer2' }
  ];

  return (
    <section className="recent-employer-section w-1/3 m-auto text-textGray">
      <div className="flex relative h-auto my-8">
        <div className="back-arrow cursor-pointer">
          <ArrowBackTwoToneIcon />
        </div>
        <h1 className="font-bold text-base title">
          Who was your most recent employer?
        </h1>
      </div>
      <div className="text-textGray mt-8">
        <input
          type="text"
          className="border outline-none bg-transparent rounded-sm w-full px-3 text-textGray input-height"
          placeholder="Company Name"
        />
      </div>

      <div className="text-textGray mt-4">
        <Select
          options={[
            { value: 'Staffing', label: 'Staffing' },
            { value: 'Employee', label: 'Employee' },
            { value: 'HR', label: 'HR' }
          ]}
          placeholder="Select your supervisor"
        />
      </div>
      <div className="text-textGray mt-4">
        <input
          type="text"
          className="border outline-none bg-transparent rounded w-full px-3 text-textGray input-height"
          placeholder="Your Title"
        />
      </div>

      <div className="flex justify-between text-textGray mt-4">
        <DatePicker label="Start Date" defaultValue="2017-05-24" />
        <DatePicker label="Send Date" defaultValue="22017-05-24" />
      </div>

      <div className="text-textGray mt-4">
        <input
          type="text"
          className="border outline-none bg-transparent rounded w-full px-3 text-textGray input-height"
          placeholder="Company phone"
        />
      </div>

      <div className="text-textGray mt-4">
        <Select isMulti options={options} placeholder="Select skills used" />
      </div>

      <div className="text-textGray mt-4">
        <input
          type="text"
          className="border outline-none bg-transparent rounded w-full px-3 text-textGray input-height"
          placeholder="Responsibilities"
        />
      </div>

      <div className="text-textGray mt-4">
        <input
          type="text"
          className="border outline-none bg-transparent rounded w-full px-3 text-textGray input-height"
          placeholder="Accomplishments"
        />
      </div>

      <div className="text-textGray mt-4">
        <input
          type="text"
          className="border outline-none bg-transparent rounded w-full px-3 text-textGray input-height"
          placeholder="Favorite project you built or contributed to?"
        />
      </div>

      <div className="flex justify-center mt-12">
        <button
          data-testid="next-button"
          className="next-btn text-white hover:bg-gray-800 font-semibold py-1 px-3 w-32 rounded-sm shadow flex justify-around"
        >
          <span className="">Next</span> <ArrowRightAltTwoToneIcon />
        </button>
      </div>
    </section>
  );
};

export default withRouter(RecentEmployer);
