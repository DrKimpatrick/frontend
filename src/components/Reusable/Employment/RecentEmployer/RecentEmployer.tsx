import React, { FC, useEffect } from 'react';
import { withRouter, useHistory } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import ArrowBackTwoToneIcon from '@material-ui/icons/ArrowBackTwoTone';
import NavBar from 'components/Reusable/Layout/NavBar/NavBar';
import { addEmployment } from 'redux/actions/employment';
import { EmploymentInput, InitialEmploymentValue } from 'components/Reusable';
import { RootState } from 'redux/store';
import { MainBackground } from 'components/Reusable/Layout/MainBackground';
import './RecentEmployer.scss';

const RecentEmployer: FC = () => {
  const dispatch = useDispatch();

  const history = useHistory();

  const reducer = useSelector((state: RootState) => {
    const { loading, errors, employment } = state.employments;
    const { message } = state.messages;
    return { message, loading, errors, employment };
  });

  useEffect(() => {
    if (reducer.message && reducer.employment) {
      history.push(`/employment-history/${reducer.employment._id}`);
    }
  }, [reducer.message, reducer.employment, history]);

  return (
    <>
      <NavBar />
      <div className="containers">
        <div className="recent-employer-section w-1/3 m-auto text-textGray">
          <div
            className="flex relative h-auto my-8"
            style={{ flexDirection: 'column' }}
          >
            <div style={{ display: 'flex' }}>
              <div
                className="back-arrow cursor-pointer"
                onClick={() => history.push('/skill-ranking')}
              >
                <ArrowBackTwoToneIcon />
              </div>
              <h1 className="font-bold text-base title">
                Who was your most recent employer?
              </h1>
            </div>

            <EmploymentInput
              initialValue={{
                companyName: '',
                title: '',
                supervisor: '',
                isCurrentPosition: false,
                responsibilities: [],
                skillsUsed: [],
                favoriteProject: '',
                accomplishments: [],
                responsibility: '',
                accomplishment: '',
                startDate: '',
                endDate: ''
              }}
              submit={(values: InitialEmploymentValue) =>
                addEmployment(values)(dispatch)
              }
              backendErrors={
                reducer.errors && reducer.errors.errors && reducer.errors.errors
              }
              loading={!!reducer.loading}
              buttonName="Next"
            />
          </div>
        </div>
      </div>
      <MainBackground />
    </>
  );
};

export default withRouter(RecentEmployer);
