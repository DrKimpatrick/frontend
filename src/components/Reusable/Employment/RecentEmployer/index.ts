import RecentEmployer from './RecentEmployer';

export { RecentEmployer };
