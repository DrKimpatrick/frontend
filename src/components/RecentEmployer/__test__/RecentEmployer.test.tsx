import { cleanup } from '@testing-library/react';
import React from 'react';
import ReactDom from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import RecentEmployer from '../RecentEmployer';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

afterEach(cleanup);
const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush
  })
}));

describe('`RecentEmployer` component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(
      <Router>
        <RecentEmployer />
      </Router>,
      div
    );
  });

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <RecentEmployer />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
