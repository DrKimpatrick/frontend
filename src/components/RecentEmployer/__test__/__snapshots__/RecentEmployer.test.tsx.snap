// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`RecentEmployer\` component matches the snapshot 1`] = `
<section
  className="recent-employer-section w-1/3 m-auto text-textGray"
>
  <div
    className="flex relative h-auto my-8"
  >
    <div
      className="back-arrow cursor-pointer"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
    </div>
    <h1
      className="font-bold text-base title"
    >
      Who was your most recent employer?
    </h1>
  </div>
  <div
    className="text-textGray mt-8"
  >
    <input
      className="border outline-none bg-transparent rounded-sm w-full px-3 text-textGray input-height"
      placeholder="Company Name"
      type="text"
    />
  </div>
  <div
    className="text-textGray mt-4"
  >
    <div
      className="outline-none placeholder-textGray placeholder-opacity-50 css-2b097c-container"
      onKeyDown={[Function]}
    >
      <div
        className=" css-gpjhn8-control"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className=" css-g1d714-ValueContainer"
        >
          <div
            className=" css-2x0uyn-placeholder"
          >
            Select your supervisor
          </div>
          <div
            className="css-b8ldur-Input"
          >
            <div
              className=""
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-autocomplete="list"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                disabled={false}
                id="react-select-4-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "boxSizing": "content-box",
                    "color": "inherit",
                    "fontSize": "inherit",
                    "label": "input",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "1px",
                  }
                }
                tabIndex="0"
                type="text"
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </div>
        </div>
        <div
          className=" css-1hb7zxy-IndicatorsContainer"
        >
          <span
            className=" css-1xjgjl1-IndicatorSeparator"
          />
          <div
            aria-hidden="true"
            className=" css-tlfecz-indicatorContainer"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <svg
              aria-hidden="true"
              className="css-6q0nyr-Svg"
              focusable="false"
              height={20}
              viewBox="0 0 20 20"
              width={20}
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="text-textGray mt-4"
  >
    <input
      className="border outline-none bg-transparent rounded w-full px-3 text-textGray input-height"
      placeholder="Your Title"
      type="text"
    />
  </div>
  <div
    className="flex justify-between text-textGray mt-4"
  >
    <div
      className="MuiFormControl-root MuiTextField-root makeStyles-textField-3"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
        data-shrink={true}
        htmlFor="date"
        id="date-label"
      >
        Start Date
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input"
          defaultValue="2017-05-24"
          disabled={false}
          id="date"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="date"
        />
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root makeStyles-textField-3"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
        data-shrink={true}
        htmlFor="date"
        id="date-label"
      >
        Send Date
      </label>
      <div
        className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiInput-input"
          defaultValue="22017-05-24"
          disabled={false}
          id="date"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          required={false}
          type="date"
        />
      </div>
    </div>
  </div>
  <div
    className="text-textGray mt-4"
  >
    <input
      className="border outline-none bg-transparent rounded w-full px-3 text-textGray input-height"
      placeholder="Company phone"
      type="text"
    />
  </div>
  <div
    className="text-textGray mt-4"
  >
    <div
      className="outline-none placeholder-textGray placeholder-opacity-50 css-2b097c-container"
      onKeyDown={[Function]}
    >
      <div
        className=" css-1px2uo9-control"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className=" css-zoz06h-ValueContainer"
        >
          <div
            className=" css-cdyrj9-placeholder"
          >
            Select skills used
          </div>
          <div
            className="css-b8ldur-Input"
          >
            <div
              className=""
              style={
                Object {
                  "display": "inline-block",
                }
              }
            >
              <input
                aria-autocomplete="list"
                autoCapitalize="none"
                autoComplete="off"
                autoCorrect="off"
                disabled={false}
                id="react-select-5-input"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                spellCheck="false"
                style={
                  Object {
                    "background": 0,
                    "border": 0,
                    "boxSizing": "content-box",
                    "color": "inherit",
                    "fontSize": "inherit",
                    "label": "input",
                    "opacity": 1,
                    "outline": 0,
                    "padding": 0,
                    "width": "1px",
                  }
                }
                tabIndex="0"
                type="text"
                value=""
              />
              <div
                style={
                  Object {
                    "height": 0,
                    "left": 0,
                    "overflow": "scroll",
                    "position": "absolute",
                    "top": 0,
                    "visibility": "hidden",
                    "whiteSpace": "pre",
                  }
                }
              >
                
              </div>
            </div>
          </div>
        </div>
        <div
          className=" css-e23gzz-IndicatorsContainer"
        >
          <span
            className=" css-1xjgjl1-IndicatorSeparator"
          />
          <div
            aria-hidden="true"
            className=" css-tlfecz-indicatorContainer"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <svg
              aria-hidden="true"
              className="css-6q0nyr-Svg"
              focusable="false"
              height={20}
              viewBox="0 0 20 20"
              width={20}
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="text-textGray mt-4"
  >
    <input
      className="border outline-none bg-transparent rounded w-full px-3 text-textGray input-height"
      placeholder="Responsibilities"
      type="text"
    />
  </div>
  <div
    className="text-textGray mt-4"
  >
    <input
      className="border outline-none bg-transparent rounded w-full px-3 text-textGray input-height"
      placeholder="Accomplishments"
      type="text"
    />
  </div>
  <div
    className="text-textGray mt-4"
  >
    <input
      className="border outline-none bg-transparent rounded w-full px-3 text-textGray input-height"
      placeholder="Favorite project you built or contributed to?"
      type="text"
    />
  </div>
  <div
    className="flex justify-center mt-12"
  >
    <button
      className="next-btn text-white hover:bg-gray-800 font-semibold py-1 px-3 w-32 rounded-sm shadow flex justify-around"
      data-testid="next-button"
    >
      <span
        className=""
      >
        Next
      </span>
       
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M16.01 11H4v2h12.01v3L20 12l-3.99-4v3z"
        />
      </svg>
    </button>
  </div>
</section>
`;
